lib/spack/spack/binary_distribution.py

```diff
@@ -599,6 +599,11 @@ def write_buildinfo_file(spec, workdir, rel=False):
 
     # Create buildinfo data and write it to disk
     buildinfo = {}
+
+    import spack.hooks.sbang as sbang
```

Comment:
Imports should be at the top of the file

Reply:
you can't import spack.hooks.sbang at the top level, for whatever reason. I don't know why, but it might have to do with the voodoo magic in how spack.hooks is setup.